modules/library/render/src/main/java/org/geotools/renderer/label/LabelPainter.java

```diff
@@ -672,19 +691,28 @@ public void paintCurvedLabel(LineStringCursor cursor) {
             }
 
             // draw halo and label
+            // extracting label first line component and compute is metrics
```

Comment:
Can there be multiple lines? If so, can subsequent lines have "worse" metrics here?

Also nit: "is metrics" -> "its metrics".